### PR TITLE
#1314 Correct singleton pattern for static compilation

### PR DESCRIPTION
**Pull request by lecourtoisflo**

### CODE REVIEW

File: dynawo/sources/Common/DYNTrace.cpp [hunk 1]
```diff
@@ -62,8 +62,6 @@ namespace keywords = boost::log::keywords;
 namespace DYN {
 
 static vector< boost::shared_ptr<Trace::TextSink> > originalSinks;
```
Comment (rosiereflo) [inline dynawo/sources/Common/DYNTrace.cpp:64]: This one remains static?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Common/DYNTrace.cpp (hunk 1 above)
Comment (lecourtoisflo) [reply dynawo/sources/Common/DYNTrace.cpp:64]: It is static at file scope, not class scope

Comment (rosiereflo) [reply dynawo/sources/Common/DYNTrace.cpp:64]: yes, but it was modified due to similar issues, so why this one is different from the others 2 in this version?

Comment (lecourtoisflo) [reply dynawo/sources/Common/DYNTrace.cpp]: reintegrated as a class member: working as well

**Review by rosiereflo — APPROVED** [review]

File: dynawo/sources/Common/DYNTraceStream.cpp [hunk 2]
```diff
@@ -67,7 +67,7 @@ TraceStream::operator<<(const char* t) {
 void
 TraceStream::flush() {
   if (buffer_) {
-    Trace::log(slv_, tag_, buffer_->str());
+    Trace::instance().log(slv_, tag_, buffer_->str());
     buffer_->str(std::string());
```
Comment (gautierbureau) [inline dynawo/sources/Common/DYNTraceStream.cpp:71]: Trace::instance() missing

**Review by gautierbureau — CHANGES_REQUESTED** [review]
I don't like how it was done and prefer the design that was done in Timers, where this a function foo that is the static version of function and that calls a foo_ inside on the instance

File: dynawo/sources/Common/DYNTraceStream.cpp (hunk 2 above)
Comment (gautierbureau) [reply dynawo/sources/Common/DYNTraceStream.cpp:71]: wrong on my part

Comment (lecourtoisflo) [reply dynawo/sources/Common/DYNTraceStream.cpp:71]: buffer_ is part of TraceStream, not Trace. But instance shouldn't be called directly and should be used to delegate the static method to the instance method

**Review by gautierbureau — APPROVED** [review]

**Pull request merged**